react/react1/week3/todo-week3/src/registration/RegistrationList.js

```diff
@@ -0,0 +1,141 @@
+import { useState, useEffect } from "react";
+import { RegistrationItem } from "./RegistrationItem";
+import { FancyBorder } from "../todoweek3/FancyBorder";
+import "./Registration.css";
+
+//main function
+export const RegistrationList = () => {
+  const [name, setName] = useState("");
+  const [startDate, setStartDate] = useState("");
+  const [startTime, setStartTime] = useState("");
+  const [endDate, setEndDate] = useState("");
+  const [endTime, setEndTime] = useState("");
+  const [update, setUpdate] = useState([]);
+  const [searchValue, setSearchValue] = useState([]);
+
+  // add info
+  const saveInfo = () => {
+    // const id = update.length + 1;
+    setUpdate((prev) => [
+      ...prev,
+      { name, startDate, endDate, startTime, endTime },
+    ]);
+  };
+  // how to calculate the time from the api ?????? with date and time together
+  useEffect(() => {
+    fetch(
+      "https://gist.githubusercontent.com/benna100/5fd674171ea528d7cd1d504e9bb0ca6f/raw"
+    )
+      .then((response) => response.json())
+      .then((data) => {
+        data.map((data, index) => {
+          setUpdate((prev) => [
+            ...prev,
+            {
+              // id: id,
+              name: data.name,
+              startDate: data.start.split("T")[0],
+              startTime: data.start.split("T")[1],
+              endDate: data.end.split("T")[0],
+              endTime: data.end.split("T")[1],
+            },
+          ]);
+        }, []);
+
+        // const apiEndtDate = data.map((data) => data.end.split("T")[1]);
+        // const apiName = data.map((data) => data.name);
+      });
+  }, []);
+  // search part
+  function searchByName() {
+    const result = update.filter((data) =>
```

Comment:
The names disappear after I seach for them. I should be able to clear the seacrh here somehow, right?

Reply:
Yes, so i add  setSearchValue("");. Is that right?